public/app/plugins/datasource/prometheus/types.ts

```diff
@@ -1,27 +1,18 @@
-import { DataQuery, DataSourceJsonData, QueryResultMeta, ScopedVars } from '@grafana/data';
+import { DataSourceJsonData, QueryResultMeta, ScopedVars } from '@grafana/data';
 
 import { PromApplication } from '../../../types/unified-alerting-dto';
 
+import { Prometheus as GenPromQuery } from './dataquery.gen';
 import { QueryEditorMode } from './querybuilder/shared/types';
 
-export interface PromQuery extends DataQuery {
-  expr: string;
-  format?: string;
-  instant?: boolean;
-  range?: boolean;
-  exemplar?: boolean;
-  hinting?: boolean;
-  interval?: string;
-  intervalFactor?: number;
-  // Timezone offset to align start & end time on backend
+export interface PromQuery extends GenPromQuery {
+  /**
+   * Timezone offset to align start & end time on backend
+   */
   utcOffsetSec?: number;
```

Comment:
might be worth checking why it is better here and not in the .cue, as backend seems to also make use of this one

Reply:
@ryantxu I want to check this after merging this PR since it is outside of the scope of this PR. But you are absolutely right. I will see if there is a reason behind this calculation being done on the frontend